src/test/java/org/codehaus/plexus/util/xml/pull/MXParserTest.java

```diff
@@ -1047,6 +1042,93 @@ public void testEncodingISO_8859_1setStringReader()
         }
     }
 
+    /**
+     * Issue 163: https://github.com/codehaus-plexus/plexus-utils/issues/163
+     *
+     * Another case of bug #163: Reader generated with ReaderFactory.newReader and the right file encoding.
```

Comment:
The file is encoded with UTF-8 **without** BOM, and XML entities are encoded in ISO-8859-1 (this is a valid combination). This is the test that reproduces the bug, which throws an exception as if the file encoding were UTF-8 **with** BOM (the invalid combination).
So as you said before, even if Maven is not affected by this, the bug in the parser is real.

Reply:
I understand this is considered a _valid combination_ by the xml spec.  But this will produce incorrect results, as the input stream is most probably decoded using UTF-8 instead of ISO-8859-1.
I'm fine with that if that's what is desired.  However, I think we can achieve both and I'll work on a fix.